ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto

```diff
@@ -334,6 +333,9 @@ message DamlContractState {
 
   // The contract key set by the contract. Optional.
   DamlContractKey contract_key = 6;
+
```

Comment:
Don't we also need to extend this structure with a field that links to DamlLogEntryId where it was created? Right now we have this "DamlLogEntryId archived_by_entry = 3;". It seems that we should have either both creattion/archival or none.

Reply:
Good point. We'll probably want to go with actual deletion of the state on archival so we shouldn't have either. What do you think about changing the interface now to support deletion of keys and dropping archived_by_entry?

Reply:
Btw, note that the "ContractId" contains the entry id in it, so no need for a "created_by_entry".

Reply:
Which ContractId do you mean? 
I don't think you mean this: https://github.com/digital-asset/daml/blob/f84e7d79d2bfcd141a6cab9f05703872b94204af/daml-lf/transaction/src/main/protobuf/com/digitalasset/daml/lf/value.proto#L107

Reply:
https://github.com/digital-asset/daml/blob/master/ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto#L297